tests/algorithms/pytorch/test_newton_raphson.py

```diff
@@ -109,6 +109,55 @@ def test_train_newton_raphson_shared_states_results(
     assert pytest.approx(np.array(expected_hessian), rel) == shared_states.hessian
 
 
+@pytest.mark.slow  # the test takes ~30s
+def test_hessian_logistic_regression_newton_raphson(torch_algo, seed):
+    """Test that the Hessian matrix is well computed for a logistic regression problem."""
+    np.random.seed(seed)
+
+    n_samples = 300
+    ndim = 100
```

Comment:
```suggestion
def test_hessian_logistic_regression_newton_raphson(torch_algo, seed):
    """Test that the Hessian matrix is well computed for a logistic regression problem."""
    np.random.seed(seed)

    n_samples = 30
    ndim = 10
```

Wdyt about setting the dim to 10 and n sample to 30 ? the test is much faster and also fail without the fix.

Reply:
ah thanks for checking! I just reproduced @jeandut 's bug that arose in this dimension and did not take the time to reduce it. It's better if it works in dim 30 indeed